packages/component-aws-s3/src/s3_utils.ts

```diff
@@ -0,0 +1,332 @@
+                Conditions: [
+                    ["content-length-range", contentOptions?.contentLengthMin ?? 1,
+                                            contentOptions?.contentLengthMax ?? 10000000], // 10MB
+                ],
+                Bucket: bucket,
+                Key: key,
+                Expires: expirationSecs,
+                Fields: {
+                    'Content-Type': contentOptions?.contentType ?? '*',
+                }
+            }
+        );
+        return {url: postPresigned.url, fields: postPresigned.fields};
+    }
+
+    static async postS3Key(awscfg: AWSServiceConfig, bucket: string, key: string, expirationSecs: number,
+        contentOptions?: {
+            contentType?: string,
+            contentLengthMin?: number,
+            contentLengthMax?: number,
+        })
+    {
+        const s3 = S3Ops.createS3Client(awscfg);
+
+        return await S3Ops.postS3KeyCmd(s3, bucket, key, expirationSecs, contentOptions);
+    }
+
+    @Communicator()
+    async postS3KeyComm(ctx: CommunicatorContext, key: string, expirationSecs: number = 1200,
+        @ArgOptional contentOptions?: {
+            contentType?: string,
+            contentLengthMin?: number,
+            contentLengthMax?: number,
+        })
+    {
+        try {
+            const cfg = this.config;
+            return await S3Ops.postS3Key(cfg.awscfg!, cfg.bucket, key, expirationSecs, contentOptions);
+        }
+        catch (e) {
+            ctx.logger.error(e);
+            throw new DBOSError(`Unable to presign post: ${(e as Error).message}`, 500);
+        }
+    }
+
+    /////////
+    // Simple Workflows
+    /////////
+
+    // Send a data string item directly to S3
+    // Use cases for this:
+    //  App code produces (or received from the user) _small_ data to store in S3 
+    //   One-shot workflow
+    //     Pick a file name
+    //     Do the S3 op
+    //     If it succeeds, write in table
+    //     If it fails drop the partial file (if any) from S3
+    // Note this will ALL get logged in the DB as a workflow parameter (and a communicator parameter) so better not be big!
+    @Workflow()
+    async saveStringToFile(ctx: WorkflowContext, fileDetails: FileRecord, content: string, @ArgOptional contentType = 'text/plain') 
+    {
+        // Running this as a communicator could possibly be skipped... but only for efficiency
+        try {
+            await ctx.invoke(this).putS3Comm(fileDetails.key, content, contentType);
+        }
+        catch (e) {
+            await ctx.invoke(this).deleteS3Comm(fileDetails.key);
+            throw e;
+        }
+    
+        await this.config.s3Callbacks?.newActiveFile(ctx, fileDetails);
+        return fileDetails;
+    }
+
+    //  App code reads a file out of S3
+    //  Just a routine, hardly worthy of a workflow
+    //    Does the DB query
+    //    Does the S3 read
+    @Workflow()
+    async readStringFromFile(ctx: WorkflowContext, fileDetails: FileRecord)
```

Comment:
If this is not doing anything else, probably take it out for now?

Reply:
Feels weird to have a write and no read.  There's going to be a temp workflow created anyway, may as well make it a real one with a name.